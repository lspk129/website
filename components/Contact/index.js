import React, { Component } from 'react';
import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import RaisedButton from 'material-ui/RaisedButton';

import ContactContainer from './ContactContainer';
import ContactInfoSection from './ContactInfoSection';
import ConsultationSection from './ConsultationSection';
import ContactTitle from './ContactTitle';
import ContactForm from './ContactForm';

const styles = {
  margin: '20px 20px',
};

class Contact extends Component {
  constructor(props) {
    super(props);
    this.state = { value: '' };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event, index, value) {
    this.setState({ value });
  }

  render() {
    return (
      <ContactContainer>
        <ContactInfoSection>
          <ContactTitle>
            Get a free consultation
          </ContactTitle>
        </ContactInfoSection>
        <ConsultationSection>
          <ContactForm>
            <TextField style={styles} floatingLabelText="Name" />
            <TextField style={styles} floatingLabelText="Your email" />
            <SelectField
              floatingLabelText="Choose a subject"
              value={this.state.value}
              onChange={this.handleChange}
              style={styles}
              fullWidth
            >
              <MenuItem value={1} primaryText="I want an introduction to your services..." />
              <MenuItem value={2} primaryText="I want a free price quote..." />
              <MenuItem value={3} primaryText="I want to ask a question" />
            </SelectField>
            {this.state.value === 3 && (
              <TextField
                floatingLabelText="Your question"
                multiLine
                rows={4}
                style={styles}
                fullWidth
              />
            )}
            <RaisedButton style={styles} label="Send" secondary />
          </ContactForm>
        </ConsultationSection>
      </ContactContainer>
    );
  }
}

export default Contact;
