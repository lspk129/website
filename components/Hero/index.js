import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import HeroImage from './HeroImage';
import HeroContainer from './HeroContainer';
import HeroTitle from './HeroTitle';
import HeroSubtitle from './HeroSubtitle';

const Hero = () => (
  <HeroImage>
    <HeroContainer>
      <HeroTitle>
        Moka Haiku:<br />web app agency
      </HeroTitle>
      <HeroSubtitle>
        Creating fast & interactive web applications with React and Meteor
      </HeroSubtitle>
      <RaisedButton label="get a free consultation" secondary />
    </HeroContainer>
  </HeroImage>
);

export default Hero;
