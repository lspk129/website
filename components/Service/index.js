import React from 'react';

import ServiceHeader from './ServiceHeader';
import ServiceContainer from './ServiceContainer';
import ServiceItem from './ServiceItem';

const Service = () => (
  <div>
    <ServiceHeader>
      What we do
    </ServiceHeader>
    <ServiceContainer>
      <ServiceItem>
        <i className="material-icons">desktop_mac</i>
        <h3>Web development</h3>
        <p>We help you build fast, functional, performance oriented websites and web apps</p>
      </ServiceItem>
      <ServiceItem>
        <i className="material-icons">color_lens</i>
        <h3>Web design</h3>
        <p>We are creating unique and adaptive designs to engage customers</p>
      </ServiceItem>
    </ServiceContainer>
  </div>
);

export default Service;
