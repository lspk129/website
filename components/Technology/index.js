import React from 'react';

import TechnologyHeader from './TechnologyHeader';
import TechnologyContainer from './TechnologyContainer';
import TechnologyItem from './TechnologyItem';

const Technology = () => (
  <div>
    <TechnologyHeader>
      We use cutting-edge technologies
    </TechnologyHeader>
    <TechnologyContainer>
      <TechnologyItem>
        <img src="/static/React_logo.png" alt="React" />
        <h3>React</h3>
      </TechnologyItem>
      <TechnologyItem>
        <img src="/static/Meteor_logo.png" alt="Meteor" />
        <h3>Meteor</h3>
      </TechnologyItem>
      <TechnologyItem>
        <img src="/static/Nextjs_logo.png" alt="Next" />
        <h3>Next.js</h3>
      </TechnologyItem>
    </TechnologyContainer>
  </div>
);

export default Technology;
