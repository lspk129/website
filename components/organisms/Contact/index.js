import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-styled-flexboxgrid';
import muiThemeable from 'material-ui/styles/muiThemeable';
import { TextField, SelectField, MenuItem, RaisedButton } from 'material-ui';
import { FaEnvelope, FaTwitter, FaGlobe } from 'react-icons/lib/fa';

import Heading from '../../atoms/Heading';
import IconWithLabel from '../../molecules/IconWithLabel';
import Division from '../../atoms/Division';
import ColStyled from '../../atoms/ColStyled';

class Contact extends Component {
  state = {
    value: '',
  };

  handleChange = (event, index, value) => this.setState({ value });

  render() {
    return (
      <Row center={'xs'}>
        <ColStyled xs={12} sm={6} color={this.props.muiTheme.palette.backgroundColor}>
          <Row center={'xs'}>
            <Col>
              <Division py={3}>
                <Division pb={1}>
                  <Heading upperCase>
                    Contact us
                  </Heading>
                </Division>
                <IconWithLabel
                  icon={<FaEnvelope color={this.props.muiTheme.palette.accent1Color} />}
                  label={'info@mokahaiku.com'}
                />
                <IconWithLabel
                  icon={<FaTwitter color={this.props.muiTheme.palette.accent1Color} />}
                  label={'@mokahaiku'}
                />
                <IconWithLabel
                  icon={<FaGlobe color={this.props.muiTheme.palette.accent1Color} />}
                  label={'www.mokahaiku.com'}
                />
              </Division>
            </Col>
          </Row>
        </ColStyled>

        <Col xs={10} sm={6}>
          <Division py={3}>
            <Division pb={1}>
              <Heading upperCase>
                Get a free consultation
              </Heading>
            </Division>
            <form>
              <Row center={'xs'}>
                <Col md={5} sm={5} xs={10}>
                  <Division px={1}>
                    <TextField floatingLabelText="Name" fullWidth />
                  </Division>
                </Col>
                <Col md={5} sm={5} xs={10}>
                  <Division px={1}>
                    <TextField floatingLabelText="Your email" fullWidth />
                  </Division>
                </Col>

                <Col xs={10}>
                  <Division px={1} textAlign={'left'}>
                    <SelectField
                      floatingLabelText="Choose a subject"
                      value={this.state.value}
                      onChange={this.handleChange}
                      fullWidth
                    >
                      <MenuItem value={1} primaryText="I want an introduction to your services..." />
                      <MenuItem value={2} primaryText="I want a free price quote..." />
                      <MenuItem value={3} primaryText="I want to ask a question" />
                    </SelectField>
                    {this.state.value === 3 && (
                    <TextField
                      floatingLabelText="Your question"
                      multiLine
                      rows={4}
                      fullWidth
                    />
                  )}
                  </Division>
                </Col>
              </Row>
              <Row center={'xs'}>
                <Col xs={10}>
                  <Division pl={1} pt={3} textAlign={'left'}>
                    <RaisedButton label={'Send'} secondary />
                  </Division>
                </Col>
              </Row>
            </form>
          </Division>
        </Col>
      </Row>
    );
  }
}

Contact.propTypes = {
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      accent1Color: PropTypes.string.isRequired,
      backgroundColor: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default muiThemeable()(Contact);
