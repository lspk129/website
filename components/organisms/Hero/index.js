import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import Container from '../../atoms/Container';
import Text from '../../atoms/Text';

const Hero = () => (
  <Container bg src={'../../../static/Hero_image.jpg'} py={4}>
    <Text tag={'h1'} hero>
      Moka Haiku:<br />web app agency
    </Text>
    <Text tag={'h2'} hero>
      Creating fast & interactive web applications with React and Meteor
    </Text>
    <Container textAlign={'center'} pt={2}>
      <RaisedButton label={'get a free consultation'} secondary />
    </Container>
  </Container>
);

export default Hero;
