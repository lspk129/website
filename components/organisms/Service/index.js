import React from 'react';
import { MdDesktopMac, MdColorLens } from 'react-icons/lib/md';
import { Row, Col } from 'react-styled-flexboxgrid';

import Card from '../../molecules/Card';
import Division from '../../atoms/Division';
import DivisionHover from '../../atoms/DivisionHover';
import Text from '../../atoms/Text';

const Service = () => (
  <Division bg src={'../../../static/Service_image.jpg'} pb={3}>
    <Division py={3}>
      <Text tag={'h2'}>What we do</Text>
    </Division>
    <Row center={'xs'}>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover backgroundColor>
          <Card
            image={<MdDesktopMac className={'icons'} size={48} />}
            label={'Web development'}
            text={'We help you build fast, functional, performance oriented websites and web apps'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover backgroundColor>
          <Card
            image={<MdColorLens className={'icons'} size={48} />}
            label={'Web design'}
            text={'We are creating unique and adaptive designs to engage customers'}
          />
        </DivisionHover>
      </Col>
    </Row>
  </Division>
);

export default Service;
