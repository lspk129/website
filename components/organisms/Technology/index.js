import React from 'react';
import { Row, Col } from 'react-styled-flexboxgrid';

import Card from '../../molecules/Card';
import Text from '../../atoms/Text';
import Division from '../../atoms/Division';
import DivisionHover from '../../atoms/DivisionHover';

const Technology = () => (
  <Division pb={3}>
    <Division py={3}>
      <Text tag={'h2'}>
        We use cutting-edge technologies
      </Text>
    </Division>
    <Row center={'xs'}>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale={1.1}>
          <Card
            image={<img src="/static/React_logo.png" alt="React" />}
            label={'React'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale={1.1}>
          <Card
            image={<img src="/static/Meteor_logo.png" alt="Meteor" />}
            label={'Meteor'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale={1.1}>
          <Card
            image={<img src="/static/Nextjs_logo.png" alt="Next.js" />}
            label={'Next.js'}
          />
        </DivisionHover>
      </Col>
    </Row>
  </Division>
);

export default Technology;
