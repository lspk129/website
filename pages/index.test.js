import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Scroll from 'react-scroll';

import IndexPage from './index';
import Hero from '../components/Hero';
import Service from '../components/Service';
import Technology from '../components/Technology';
import Contact from '../components/Contact';

describe('IndexPage', () => {
  const shallowWrapper = shallow(<IndexPage />);

  it('renders correctly', () => {
    const tree = toJson(shallowWrapper);
    expect(tree).toMatchSnapshot();
  });

  it('renders IndexPage component', () => {
    expect(shallowWrapper).toBePresent();
  });

  it('renders Hero component', () => {
    expect(shallowWrapper.find(Hero)).toBePresent();
  });

  it('renders Service componant', () => {
    expect(shallowWrapper.find(Service)).toBePresent();
  });

  it('renders Technology component', () => {
    expect(shallowWrapper.find(Technology)).toBePresent();
  });

  it('renders Scroll.Element component with name link "Service"', () => {
    expect(shallowWrapper.find(Scroll.Element).at(0)).toBePresent();
    expect(shallowWrapper.find(Scroll.Element).at(0)).toHaveProp('name', 'Service');
  });

  it('renders Scroll.Element component with name link "Technology"', () => {
    expect(shallowWrapper.find(Scroll.Element).at(1)).toBePresent();
    expect(shallowWrapper.find(Scroll.Element).at(1)).toHaveProp('name', 'Technology');
  });

  it('renders Contact component', () => {
    expect(shallowWrapper.find(Contact)).toBePresent();
  });
});
