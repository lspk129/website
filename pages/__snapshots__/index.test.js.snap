// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="sc-bdVaJa bcrzSP"
>
  <h1
    className="sc-bwzfXH jQXNWG"
  >
    Moka Haiku:
    <br />
    web app agency
  </h1>
  <h2
    className="sc-htpNat gINAtw"
  >
    Creating fast & interactive web applications with React and Meteor
  </h2>
  <button
    className="sc-bxivhb jxYIva"
  >
    get a free consultation
  </button>
</div>
`;
